pkg/runtime/wfengine/workflow.go

```diff
@@ -387,11 +391,13 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 	case <-ctx.Done(): // caller is responsible for timeout management
 		return ctx.Err()
 	case completed := <-callback:
+		// [Question] Is this rescheduling the workflow to be completed ?
 		if !completed {
 			return newRecoverableError(errExecutionAborted)
 		}
 	}
 
+	// [Question] Not quite sure why this below logic is written.
```

Comment:
The comment below explains the logic. when `continueAsNew` is invoked, then workflow resets it's history etc. and runs further. It's like a new workflow. So the previous `generation` shouldn't impact this one. e.g. Timers of previous generation don't get triggered in new one: https://github.com/dapr/dapr/blob/HEAD/pkg/runtime/wfengine/workflow.go#L309-L310

Reply:
Understood, what will be the use-case for it though ?